src/registry/artifactory.rs

```diff
@@ -197,6 +200,20 @@ impl Artifactory {
 
     /// Publishes a package to artifactory
     pub async fn publish(&self, package: Package, repository: String) -> miette::Result<()> {
+        // abort publishing if we have local dependencies
+        if let Some(local) = package
+            .manifest
+            .dependencies
+            .iter()
+            .find(|d| d.manifest.is_local())
+        {
+            return Err(miette!(
+                "unable to public {} to artifactory due to local dependency {}",
+                package.name(),
+                local.package
+            ));
+        }
```

Comment:
we could also print the total number of local deps to avoid people forgetting one, getting the error again and restarting the cycle

Reply:
Good hint! Having to rerun buffrs publish over and over just to get a new error every time is a bad ux!